src/storage/src/source/postgres/replication.rs

```diff
@@ -738,14 +769,23 @@ fn extract_transaction<'a>(
             };
             metrics.total.inc();
             match message {
-                Insert(body) if errored_tables.contains(&body.rel_id()) => continue,
-                Update(body) if errored_tables.contains(&body.rel_id()) => continue,
-                Delete(body) if errored_tables.contains(&body.rel_id()) => continue,
-                Relation(body) if errored_tables.contains(&body.rel_id()) => continue,
+                Insert(body) if !table_info.contains_key(&body.rel_id()) => continue,
+                Update(body) if !table_info.contains_key(&body.rel_id()) => continue,
+                Delete(body) if !table_info.contains_key(&body.rel_id()) => continue,
+                Relation(body) if !table_info.contains_key(&body.rel_id()) => continue,
                 Insert(body) => {
                     metrics.inserts.inc();
                     let row = unpack_tuple(body.tuple().tuple_data());
-                    yield (body.rel_id(), row, 1);
+                    let rel = body.rel_id();
+                    for ((output, _), row) in table_info
+                        .get(&rel)
+                        .map(|o| o.iter().filter(|(o, _)| !errored_outputs.contains(o)))
+                        .into_iter()
+                        .flatten()
+                        .repeat_clone(row)
+                    {
+                        yield (rel, *output, row, 1);
```

Comment:
Something doesn't sit right with me with having the cloning logic happen in here. Can this not be lifted higher up or even be a separate timely operator after having produced the initial raw collection?

Reply:
I see what you're saying about `exact_transaction`, there should be a way for me to refactor this so this method operates on a table-basis and we do the cloning for multiple source exports one level up from this when necessary

Reply:
Ah now I remember why I had to refactor `extract_transaction` to be per-export -- it's because the when we received a `Relation` message (a DDL change) for a given table, we need to verify the new schema against our current one to determine that it's compatible. Because schemas can be different between source-exports, that needs to be done per-export. That logic is all currently performed within `extract_transaction` since it determines whether it should yield an error row or not